### ISSUE TITLE
Running but not logging or reporting

### ISSUE THREAD

**Issue by pnotequalnp** [cdown/psi-notify#20]
On Arch after installing via the AUR package, running `psi-notify` gives the following:
```
~/ > psi-notify
INFO: Using system-global resource pressures.
INFO: Config:

      Log pressures: true
      Update interval: 5s

      Thresholds:
        - CPU avg10 some: 50.00
        - Memory avg10 some: 10.00
        - I/O avg10 full: 15.00

INFO: Pressure monitoring started.
WARN: PSI dir (logind seat) seems to have gone away, reopening
```
And nothing is ever logged, no notifications are sent. The systemd service does the same (as expected).

**Comment by cdown** [comment]
~What's unexpected here? If you want continual logging of pressures, you need `log_pressures yes` in your config. Notifications will only be sent when the thresholds are breached.~

Oh, I see it's already `yes`.

**Comment by cdown** [comment]
Clearly there's an inversion in the using_seat check, though, although it's unrelated to your question:

```
INFO: Using system-global resource pressures.
[...]
WARN: PSI dir (logind seat) seems to have gone away, reopening
```

**Comment by cdown** [comment]
Oh, I see what you're saying, you have `Log pressures: true` but nothing gets printed.

That's interesting and probably indicates we entered some pathological state in psi-notify. It is particularly weird that it immediately complains that the PSI dir went away, that means that the /proc/pressure dir disappeared. But then it clearly came back, because we successfully opened it again, or we would die straight afterwards...

Do you have some kind of special /proc setup? Is this reproducible every time? If so, can you run for 30 seconds or so with `strace -t -f -o strace.out psi-notify` and show me the output?

**Comment by cdown** [comment]
Ok, I've got a repro. :-)

**Comment by cdown** [comment]
Thanks for the report, the fix is in 820b0255e77b3a26f4dd1e9c45e66aa13e4468ca. This primarily affects cgroup v1 users. The issue is bracket hell inside the dir_fd check in get_psi_dir_fd (which I'm kind of amazed clang didn't complain about, but...).

That function should be reworked, I'll bear that in mind for cleanups in the near future. :-)

psi-notify will actually continue to work and send notifications, but since stdout is inadvertently closed, it can't write about it:

```
read(4, "some avg10=0.00 avg60=0.00 avg30"..., 128) = 110
write(1, "INFO: Current I/O pressures: som"..., 68) = -1 EBADF (Bad file descriptor)
write(1, "INFO: Current I/O pressures: ful"..., 68) = -1 EBADF (Bad file descriptor)
```

**Comment by cdown** [comment]
1.2.1 tagged and released to AUR. Thank you for the report, again!
